python/whylogs/experimental/api/logger/__init__.py

```diff
@@ -333,6 +353,18 @@ def log_batch_ranking_metrics(
     hit_ratio = formatted_data["count_at_k"].apply(lambda x: bool(x)).sum() / len(formatted_data)
     mrr = (1 / formatted_data["top_rank"]).replace([np.inf, np.nan], 0)
     output_data["reciprocal_rank"] = mrr
+
+    if schema and schema.segments:
+        formatted_data = data.copy(deep=True)  # TODO: does this have to be deep?
+        if log_full_data:
+            return log(pandas=pd.concat([formatted_data, output_data], axis=1), schema=schema)
+        else:
+            segment_columns = _get_segment_columns(schema, formatted_data)
+            segmentable_data = formatted_data[segment_columns]
+            result = log(pandas=pd.concat([segmentable_data, output_data], axis=1), schema=schema)
+            result = _drop_non_output_columns(result, set(output_data.columns))
+            return result
+
```

Comment:
This skips `accuracy_k` for segmented schemas. I can drop it for the unsegmented case or in a follow-up PR.

Reply:
@richard-rogers let's drop accuracy_k for both segmented and unsegmented profiles please.